### PR TITLE
Add a new SyclEventRaw class

### PR DESCRIPTION
**Pull request by vlad-perevezentsev**
This PR will add a new `SyclEventRaw` class and tests for it.

### CODE REVIEW

File: dpctl/_sycl_event.pyx [hunk 1]
```diff
@@ -64,3 +73,132 @@ cdef class SyclEvent:
...
+    cdef DPCTLSyclEventRef ERef = NULL
+    if pycapsule.PyCapsule_IsValid(o, "SyclEventRef"):
+        ERef = <DPCTLSyclEventRef> pycapsule.PyCapsule_GetPointer(
+            o, "SyclEventRef"
+        )
+        DPCTLEvent_Delete(ERef)
+
+
+cdef class _SyclEventRaw:
+    """ Python wrapper class for a cl::sycl::event.
```
Comment (oleksandr-pavlyk) [inline dpctl/_sycl_event.pyx:87]:
```suggestion
    """ Python wrapper class for a ``cl::sycl::event``.
```

File: dpctl/_sycl_event.pyx [hunk 2]
```diff
@@ -64,3 +73,132 @@ cdef class SyclEvent:
...
+    """
+
+    def __dealloc__(self):
+        DPCTLEvent_Delete(self._event_ref)
+
+
+cdef class SyclEventRaw(_SyclEventRaw):
+    """ Python wrapper class for a cl::sycl::event.
```
Comment (oleksandr-pavlyk) [inline dpctl/_sycl_event.pyx:95]:
```suggestion
    """ Python wrapper class for a ``cl::sycl::event``.
```

File: dpctl/_sycl_event.pyx [hunk 3]
```diff
@@ -64,3 +73,132 @@ cdef class SyclEvent:
...
+            self._event_ref = ERef_copy
+            return 0
+        else:
+            return -128
+
+    def __cinit__(self,arg=None):
+        cdef int ret = 0
+        if arg == None:
+            ret = self._init_event_default()
+        elif isinstance(arg, _SyclEventRaw):
```
Comment (oleksandr-pavlyk) [inline dpctl/_sycl_event.pyx:151]: We probably want here `type(arg) is _SyclEventRaw`.

File: dpctl/_sycl_event.pyx [hunk 4]
```diff
@@ -64,3 +73,132 @@ cdef class SyclEvent:
...
+            elif (ret == -3):
+                raise ValueError(
+                    "Internal Error: Could not create a copy of a sycl event."
+                )
+            raise ValueError(
+                "Unrecognized error code ({}) encountered.".format(ret)
+            )
+
+    cdef DPCTLSyclEventRef get_event_ref(self):
+        """ Returns the DPCTLSyclEventRef pointer for this class.
```
Comment (oleksandr-pavlyk) [inline dpctl/_sycl_event.pyx:178]:
```suggestion
        """ Returns the `DPCTLSyclEventRef` pointer for this class.
```

File: dpctl/_sycl_event.pyx [hunk 5]
```diff
@@ -64,3 +73,132 @@ cdef class SyclEvent:
...
+        """
+        return self._event_ref
+
+    cpdef void wait(self):
+        DPCTLEvent_Wait(self._event_ref)
+
+    def addressof_ref(self):
+        """ Returns the address of the C API DPCTLSyclEventRef pointer as
```
Comment (oleksandr-pavlyk) [inline dpctl/_sycl_event.pyx:186]:
```suggestion
        """ Returns the address of the C API `DPCTLSyclEventRef` pointer as
```

File: dpctl/_sycl_event.pyx [hunk 6]
```diff
@@ -64,3 +73,132 @@ cdef class SyclEvent:
...
+        a size_t.
+
+        Returns:
+            The address of the DPCTLSyclEventRef object used to create this
```
Comment (oleksandr-pavlyk) [inline dpctl/_sycl_event.pyx:190]: Same here

File: dpctl/_sycl_event.pyx [hunk 7]
```diff
@@ -64,3 +73,132 @@ cdef class SyclEvent:
...
+            SyclEvent cast to a size_t.
```
Comment (oleksandr-pavlyk) [inline dpctl/_sycl_event.pyx:191]:
```suggestion
            `SyclEvent` cast to a size_t.
```

File: dpctl/_sycl_event.pyx [hunk 8]
```diff
@@ -64,3 +73,132 @@ cdef class SyclEvent:
...
+        """
+        return int(<size_t>self._event_ref)
```
Comment (oleksandr-pavlyk) [inline dpctl/_sycl_event.pyx:193]: Call to `int` constructor is not needed here.

**Review by oleksandr-pavlyk — CHANGES_REQUESTED** [review]
Minor changes needed, but looks great!

File: dpctl/_sycl_event.pyx (hunk 3 above)
Comment (vlad-perevezentsev) [reply dpctl/_sycl_event.pyx:151]: `isinstance` caters for inheritance while `type` equality checking does not. Don't we need to consider `SyclEventRaw` class when copying?

Comment (oleksandr-pavlyk) [reply dpctl/_sycl_event.pyx:151]: We do not do so for other classes, `SyclDevice`, `SyclQueue`, etc.

Perhaps we need to introduce `.copy()` methods to all of them.

File: dpctl/_sycl_event.pyx [hunk 9]
```diff
@@ -64,3 +73,132 @@ cdef class SyclEvent:
...
+cdef void _event_capsule_deleter(object o):
+    cdef DPCTLSyclEventRef ERef = NULL
+    if pycapsule.PyCapsule_IsValid(o, "SyclEventRef"):
+        ERef = <DPCTLSyclEventRef> pycapsule.PyCapsule_GetPointer(
+            o, "SyclEventRef"
+        )
+        DPCTLEvent_Delete(ERef)
+
+
+cdef class _SyclEventRaw:
```
Comment (oleksandr-pavlyk) [inline dpctl/_sycl_event.pyx:86]: A short docstring is needed.

**Comment by oleksandr-pavlyk** [comment]
@vlad-perevezentsev The Windows Py3.7 build failed because there is not OpenCL CPU driver installed on that machine:

```
[05:13:44] :	 [Step 2/2] ..\_test_env\lib\site-packages\dpctl\tests\test_sycl_event.py:41: 
[05:13:44] :	 [Step 2/2] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
[05:13:44] :	 [Step 2/2] >   ???
[05:13:44] :	 [Step 2/2] E   dpctl._sycl_queue.SyclQueueCreationError: SYCL Device 'opencl:cpu' could not be created.
[05:13:44] :	 [Step 2/2] dpctl\_sycl_queue.pyx:327: SyclQueueCreationError
[05:13:44] :	 [Step 2/2] ---------------------------- Captured stderr call -----------------------------
[05:13:44] :	 [Step 2/2] Could not find a device that matches the specified filter(s)! -1 (CL_DEVICE_NOT_FOUND)
```

Could you please add logic to skip such a test if the queue creation did not succeed? Please look at other tests (in test file for queue submit) for examples.

**Review by oleksandr-pavlyk — APPROVED** [review]

**Pull request merged**